src/domain/products/product-form/sections/sales-channels.tsx

```diff
@@ -19,46 +23,83 @@ const SalesChannels: React.FC<SalesChannelsProps> = ({
   isEdit = false,
   product,
 }) => {
-  const { count } = useAdminSalesChannels()
-  const remainder = Math.max(product?.sales_channels?.length - 3, 0)
+  const {
+    state: isScModalOpen,
+    open: openScModal,
+    close: closeScModal,
+  } = useToggleState()
+
+  const { salesChannels, setSalesChannels } = useProductForm()
+
+  const { count, isLoading: isLoadingSalesChannels } = useAdminSalesChannels()
+  const { store, isLoading: isLoadingStore } = useAdminStore()
 
+  const isLoading = isLoadingSalesChannels || isLoadingStore
+
+  useEffect(() => {
+    if (!isEdit && !isLoadingStore) {
+      setSalesChannels([store?.default_sales_channel as SalesChannel])
+    }
+  }, [isLoadingStore])
+
+  const remainder = Math.max(product?.sales_channels?.length - 3, 0)
   return (
-    <BodyCard
-      title="Sales Channels"
-      subtitle="This product will only be available in the default sales channel if left untouched."
-      className="min-h-[200px]"
-    >
-      <div className="flex space-x-2">
-        <div className="flex space-x-2 max-w-[600px] overflow-clip">
-          {isEdit &&
-            product.sales_channels
-              ?.slice(0, 3)
-              .map((sc) => <SalesChannelBadge channel={sc} />)}
-        </div>
-        {remainder > 0 && (
-          <Badge variant="ghost">
-            <div className="flex items-center h-full inter-base-regular text-grey-50">
-              + {remainder} more
+    <>
+      <BodyCard
+        title="Sales Channels"
+        subtitle="This product will only be available in the default sales channel if left untouched."
+        className="min-h-[200px]"
+      >
+        {isLoading ? (
+          <Spinner />
+        ) : (
+          <>
+            <div className="flex space-x-2">
+              <div className="flex space-x-2 max-w-[600px] overflow-clip">
+                {salesChannels?.slice(0, 3).map((sc) => (
+                  <SalesChannelBadge channel={sc} />
+                ))}
+              </div>
+              {remainder > 0 && (
+                <Badge variant="ghost">
+                  <div className="flex items-center h-full inter-base-regular text-grey-50">
+                    + {remainder} more
+                  </div>
+                </Badge>
+              )}
             </div>
-          </Badge>
-        )}
-      </div>
-      <span className="inter-base-regular text-grey-50 mb-large mt-base">
-        Available in{" "}
-        <span className="inter-base-semibold text-grey-90">
-          {isEdit ? product.sales_channels?.length : 0}
-        </span>{" "}
-        out of <span className="inter-base-semibold text-grey-90">{count}</span>{" "}
-        Sales Channels
-      </span>
+            <span className="inter-base-regular text-grey-50 mb-large mt-base">
+              Available in{" "}
```

Comment:
question: is that `{" "}` equivalent to `&nbsp;`?

Reply:
Almost, `{" "}` is a regular space `&nbsp` is a non-breaking, non-collapsible space